src/App.tsx

```diff
@@ -1,61 +1,122 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [selectedUserID, setSelectedUserID] = useState(0);
+  const [title, setTitle] = useState('');
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+
+  const [isTitleEntred, setIsTitleEntred] = useState(true);
+  const [isUserChoosed, setIsUserChoosed] = useState(true);
+
+  const handleUpdatingTodos = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+
+    const selectedUser = getUserById(selectedUserID);
+
+    const todoID: number = Math
+      .max(...visibleTodos.map(todo => todo.id)) + 1;
```

Comment:
is map a mutating array method?

Reply:
oops... :woman-facepalming:

Reply:
I don't make a copy with spred operator... I just unpack the array so that its elements can be used inside the Math.max))
Or I'm not right?